tests/core/test_repo.py

```diff
@@ -1,179 +1,190 @@
+        _ = Repository(str(tmp_path)).init_repo(
+            repo_path=tmp_path, analytics_opt_in=False
+        )
+
+
+def test_repo_init_without_git_repo_initialized_raises_error(
+    tmp_path: str,
+) -> None:
+    """Check initializing repository without git repository raises error"""
+    with pytest.raises(Exception):
+        _ = Repository(str(tmp_path)).init_repo(
+            repo_path=tmp_path, analytics_opt_in=False
+        )
+
+
+def test_init_repo_creates_a_zen_folder(tmp_path: str) -> None:
+    """Check initializing repository creates a ZenML folder"""
+    _ = Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    local_stack = LocalService().get_stack("local_stack")
+    repo.init_repo(
+        repo_path=tmp_path, analytics_opt_in=False, stack=local_stack
+    )
+    assert os.path.exists(os.path.join(tmp_path, ZENML_DIR_NAME))
+
+
+def test_get_git_wrapper_returns_the_wrapper(tmp_path: str) -> None:
+    """Check get_git_wrapper returns the wrapper"""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    assert repo.get_git_wrapper() is not None
+    assert repo.get_git_wrapper().git_repo == GitWrapper(tmp_path).git_repo
+
+
+def test_getting_the_active_service_returns_local_service(
+    tmp_path: str,
+) -> None:
+    """Check getting the active service"""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    assert repo.get_service() is not None
+    assert isinstance(repo.get_service(), BaseComponent)
+    assert isinstance(repo.get_service(), LocalService)
+    assert repo.get_service() == repo.service
+
+
+def test_getting_active_stack_key_returns_local_stack(
+    tmp_path: str,
+) -> None:
+    """Check getting the active stack key"""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    repo.set_active_stack("local_stack")
+    assert repo.get_active_stack_key() == "local_stack"
+
+
+def test_getting_active_stack_returns_local_stack(
+    tmp_path: str,
+) -> None:
+    """Check getting the active stack"""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    repo.set_active_stack("local_stack")
+    assert repo.get_active_stack() == LocalService().get_stack("local_stack")
+    assert isinstance(repo.get_active_stack(), BaseStack)
+
+
+def test_get_pipelines_returns_list(tmp_path: str) -> None:
+    """Check get_pipelines returns a list"""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    repo.set_active_stack("local_stack")
+    our_pipelines = repo.get_pipelines()
+    assert our_pipelines is not None
+    assert isinstance(our_pipelines, list)
+
+
+def test_get_pipelines_returns_same_list_when_stack_specified(tmp_path) -> None:
+    """Check get_pipelines returns the same list when stack specified"""
+    Repo.init(tmp_path)
+    repo = Repository(str(tmp_path))
+    repo.set_active_stack("local_stack")
+    our_pipelines_default = repo.get_pipelines()
+    our_pipelines_local = repo.get_pipelines(stack_key="local_stack")
```

Comment:
I think in this case this might both simply be an empty list, we should update this test once we have some setup of an isolated testing environment in which we can run pipelines (or some kind of Mock Metadata Store would also be possible I guess)